lib/clAdapters/google-mail/index.js

```diff
@@ -229,21 +271,18 @@ var convertEmailListToArrayOfEmailAddressObjects = function(emailList) {
 };
 
 var hasLabel = function(message, labelValue) {
-  return message.labelIds.indexOf(labelValue) >= 0;
+  return message.labelIds && message.labelIds.length && (message.labelIds.indexOf(labelValue) >= 0);
 };
 
 var mapEmailData = function(emailData) {
   var mappedData = [];
 
   for (var userIter = 0; userIter < emailData.length; userIter++) {
-    var mappedUser = {
-      email: emailData[userIter].email,
-      filterStartDate: emailData[userIter].filterStartDate,
-      filterEndDate: emailData[userIter].filterEndDate,
+    var mappedUser = _.assign({}, emailData[userIter], {
       data: [],
       success: emailData[userIter].success,
       errorMessage: emailData[userIter].errorMessage
-    };
+    });
 
     if (emailData[userIter].success) {
       for (var i = 0; i < emailData[userIter].data['messages'].length; i++) {
```

Comment:
Can you take me off this distro?  Thanks.

Tom Maffey
CrossLeader.
McChrystal Group
804 426 5647


On Jan 4, 2016, at 19:09, Ben Southgate <notifications@github.com<mailto:notifications@github.com>> wrote:


In lib/clAdapters/google-mail/index.js<https://github.com/CrossLead/crosslead-adapters/pull/15#discussion_r48799131>:

>        data: [],
>        success: emailData[userIter].success,
>        errorMessage: emailData[userIter].errorMessage
> -    };
> +    });
>
>      if (emailData[userIter].success) {
>        for (var i = 0; i < emailData[userIter].data['messages'].length; i++) {


Might want to cache the length prop here as this is several properties deep and in an inner loop so might get expensive

-
Reply to this email directly or view it on GitHub<https://github.com/CrossLead/crosslead-adapters/pull/15/files#r48799131>.